lib/claim/token-holder-revenue-fund-contracts-ensemble.js

```diff
@@ -0,0 +1,359 @@
+            span.startBlock =
+                (await configuration.contract.closedAccrualsByCurrency(
+                    currency.ct.toString(), currency.id, span.startAccrual
+                )).startBlock.toNumber();
+            span.endBlock =
+                (await configuration.contract.closedAccrualsByCurrency(
+                    currency.ct.toString(), currency.id, span.endAccrual
+                )).endBlock.toNumber();
+
+            if (!configuration.spansByCurrency)
+                configuration.spansByCurrency = new Map();
+
+            configuration.spansByCurrency.set(currencyKey, span);
+
+            globalAccrualOffset = span.endAccrual + 1;
+        }
+
+        _spannedByCurrency.get(this).set(currencyKey, true);
+    }
+
+    /**
+     * Return status of whether the given accrual has been fully claimed
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @returns {boolean} True if fully claimed, else false
+     */
+    async fullyClaimed(wallet, currency, accrual) {
+        if (!this.isSpanned(currency))
+            await this.span(currency);
+
+        const ensemble = _ensemble.get(this);
+        const currencyKey = currencyToKey(currency);
+
+        for (const configuration of ensemble) {
+            const span = configuration.spansByCurrency.get(currencyKey);
+            if (span.startAccrual <= accrual && accrual <= span.endAccrual) {
+                return configuration.contract.fullyClaimed(
+                    wallet.address, currency.ct.toString(), currency.id, accrual
+                );
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the count of closed accruals for the given currency
+     * @param {Currency} currency - The currency
+     * @returns {Promise} A promise that resolves into the count of closed accruals
+     */
+    async closedAccrualsCount(currency) {
+        const ensemble = _ensemble.get(this);
+
+        const count = (await Promise.all(ensemble.map(c => c.contract.closedAccrualsCount(
+            currency.ct.toString(), currency.id
+        ))))
+            .reduce((a, c) => a.add(c), bigNumberify(0));
+
+        return count;
+    }
+
+    /**
+     * Get claimable amount for the given span of accruals
+     * @param {Wallet} wallet - The claimer nahmii wallet
+     * @param {Currency} currency - The currency
+     * @param {number} startAccrual - The lower accrual index boundary
+     * @param {number} endAccrual - The upper accrual index boundary
+     * @returns {Promise} A promise that resolves into a BigNumber value representing the claimable amount.
+     */
+    async claimableAmountByAccruals(wallet, currency, startAccrual, endAccrual) {
+        if (startAccrual > endAccrual)
+            throw new Error(`Ordinality mismatch of startAccrual > endAccrual (${startAccrual} > ${endAccrual})`);
+
+        if (!this.isSpanned(currency))
```

Comment:
Candidate for removal?

Reply:
Being replaced by `if (!isSpanned.call(this, currency))`.